File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    multiple = []
    horizontal = 0
    vertical = 0
    for i in range(1,yellow+1):
        if yellow % i == 0:
            multiple.append(i)
    for j in multiple:
        temp = yellow / j
        horizontal = j + 2
        vertical = temp + 2
        if brown == horizontal * 2 + (vertical - 2) * 2:
            answer.append(vertical)
            answer.append(horizontal)
            break
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let sum = brown + yellow;
    
    for(let i=1; i<=brown/2; i++){
        if( i * ((brown - i * 2) / 2 + 2 ) == sum){
            return [i, (brown - i *2) /2 + 2].sort((a,b) => b-a);
        }
    }
}